### ISSUE TITLE
Right-click causes App to crash

### ISSUE THREAD

**Issue by eightyfive** [Automattic/wp-desktop#241]
Hi,

Thanks for a great Product. I'm facing a rather annoying problem, when I right click inside a Post edit, the app goes blank, it seems to freeze and ultimately I quit it (Cmd + Q) because it doesn't seem to come back (and I don't have the patience to wait for it to come back maybe).

My questions are:
1. Does the blank screen means it's gone or does it only mean it is processing things? (and I would have to wait for the App to come back)
2. Is there any way to debug or know what happens when the App goes blank?
3. Is there a log/trace/debug console that I can look at?
4. Do you have any ideas what actions the right click triggers?

Thanks in advance,
Ben

PS: I am running the latest version, 1.8.0.

**Comment by rachelmcr** [comment]
Thanks for the reporting this! I can reproduce this issue on version 1.8.0. When I right-click in the editor, the following errors appear in the console:

```
ipc-renderer.js:10 'window.webkitStorageInfo' is deprecated. Please use 'navigator.webkitTemporaryStorage' or 'navigator.webkitPersistentStorage' instead.
ipc-renderer.js:10 'webkitIndexedDB' is deprecated. Please use 'indexedDB' instead.
```

(You can open developer tools in the app using CTRL+SHIFT+i on a Mac.)

I've confirmed that this only happens in the Visual editor — I can right-click elsewhere on the screen, in the post title field, and in the HTML editor without any issue.

**Comment by sendhil** [comment]
Yuck, I can repro as well. Will investigate.

**Comment by circlecast** [comment]
This issue is happening on the Windows version also. Exact same results Right click the screen freezes for a short bit. Then the screen goes blank, except for the menu item at the top you are able to still use the File Edit Window and Help menus. You then have to quit the program and restart it.

**Comment by lizkarkoski** [comment]
Related reports in the WordPress.com support forums:
http://en.forums.wordpress.com/topic/crashing-9

nicooptical using Windows 10 desktop app version 1.8.0 
peachtreegolfer on Mac using desktop app version 1.9.0 - They have uninstalled, restarted, reinstalled the app, with the issue persisting. 

Each user describes the app as crashing when they right click within the app.

**Comment by rwaynick** [comment]
Also reported in #519404-chat,

> I have noticed that since version 1.8 when copying and pasting the app freezes. I have narrowed this down to right clicking anywhere within a page or blog post. this requires a forced quit, as the window goes white. OSX 10.12.1

**Comment by Nicholas-Leslie** [comment]
All users granted permission running for Window 8 updated to Version 1.9.0 still going blank on right mouse click.

[Window Title]
Error

[Main Instruction]
WordPress.com ran into an error

[Content]
Please restart the app and try again.

If you continue to have issues, please contact us at help@wordpress.com and mention the error details below:

TypeError: Error processing argument at index 0, conversion failure from C:\Program Files (x86)\WordPress.com\resources\app\public_desktop\win10-tray-icon.ico
    at TypeError (native)
    at new WindowsPlatform (C:\Program Files (x86)\WordPress.com\resources\app\desktop.js:143816:15)
    at Platform.setMainWindow (C:\Program Files (x86)\WordPress.com\resources\app\desktop.js:5558:20)
    at showAppWindow (C:\Program Files (x86)\WordPress.com\resources\app\desktop.js:143995:12)
    at C:\Program Files (x86)\WordPress.com\resources\app\desktop.js:144004:15
    at Server.<anonymous> (C:\Program Files (x86)\WordPress.com\resources\app\desktop.js:144085:4)
    at Server.g (events.js:286:16)
    at emitNone (events.js:86:13)
    at Server.emit (events.js:185:7)
    at emitListeningNT (net.js:1279:10)

System info: {"platform":"win32","release":"10.0.14393","version":"1.9.0","build":"release"}

[OK]

Settings applied to: WordPress.com:
Windows compatibility mode: Windows 8

Settings applied to: WordPress.com:
User Account Control:  Run as administrator

Still freezes on right click Windows 10, even setting .exe file to run as administrator.

**Comment by chaitanyamsv** [comment]
Reported by an user in #2934134-t. 

I reproduced this bug in Mac app v 2.0.0.

**Comment by liviopv** [comment]
A user reported a similar bug in the Linux app as well.

User's setup details: https://mountainrepublic.files.wordpress.com/2016/11/support-details-tech-support-management.png

> When I right-click in the "visual" side of the text editor, the app freezes up and I have to close and restart it. This has been a problem over the last several versions of the app. It would be nice if the ability to copy and paste into the body of the "visual" side of the editor was restored.

**Comment by rachelmcr** [comment]
I confirmed that this is still an issue in the latest beta (2.1.0-beta3). @sendhil any updates from your investigation? Let me know if there's anything I can do to help.

**Comment by markoskin** [comment]
"me too".  2.2.0-beta1 still does it.   I moved off of "stable" to "beta" hoping it would fix this issue.  NO luck.

**Comment by gracie** [comment]
Reported in #574147-chat

Desktop App version: 2.2.0-beta1

OSX Version: 10.12.3 beta

> Hi,
> Whenever I right-click the Wordpress.com macOS app it crashes. The app goes white indefinitely and I have to force quit and re-launch it to recover.
> I've tested on two different MacBook pros and get the same outcome.
> App version is 2.2.0-beta1.
> I've kind of trained myself not to right-click ever while using the beta.
> Cheers!
>

**Comment by cagrimmett** [comment]
+1
Wordpress.com 2.2.0
macOS 10.12.3 Beta (16D25a)

Also +1 on Windows 2.2.0 - https://en.forums.wordpress.com/topic/right-click-bug?replies=2#post-2837687

**Comment by hiramknick** [comment]
I have reproduced this issue as well.  Right clicking within the application causes it to crash requiring a force-quit (MacOSX).  I tried to upgrade to 2.2.0 as well, is not a fix.

**Comment by lizkarkoski** [comment]
A Mac user in the forums reached out to us to report the right click app crash issue as well: https://en.forums.wordpress.com/topic/app-crashes-1?replies=1

**Comment by markl181** [comment]
Any news on this issue?

**Comment by mikejolley** [comment]
This happens on my macs too - 2.2.0

**Comment by SilentWolf101** [comment]
I too have this issue, but for me, simply quitting and relaunching produces the same issue. I have to completely uninstall and reinstall every single time I accidentally right-click the screen (which is a lot since my cat likes to sit on my mouse).

**Comment by lancewillett** [comment]
I've experienced this inside the post editor, using the Mac desktop app version 2.3.0-beta3. When you highlight a word, and right-click — you lose capability to use the app. It doesn't crash (as in close or die) but the UI becomes unresponsive.

**Comment by supernovia** [comment]
We have another instance [here](http://en.forums.wordpress.com/topic/mac-app-freezes-when-i-right-click).

**Comment by nusagates** [comment]
i got same issue today. i am running Wp Desktop on Windows 10 Education version 64 bit.

**Comment by markl181** [comment]
Unfortunately not fixed in the 2.4.0 update

**Comment by alexjgustafson** [comment]
3220537-t

**Comment by DrPuppet** [comment]
Still not fixed with 2.5.0 update either.

**Comment by KokkieH** [comment]
Another report in 861538-f, and I confirmed myself as well.

**Comment by rachelmcr** [comment]
I confirmed this is fixed in the 2.6.0 beta. Once 2.6.0 is released, please let us know if you experience a crash when you right-click in the editor.
